### PR TITLE
[CLOUDTRUST-2937] Editing PII must void accreditations (add missing fields)

### PR DESCRIPTION
**Pull request by fperot74**

### CODE REVIEW

File: pkg/management/component.go [hunk 1]
```diff
@@ -470,7 +470,16 @@ func (c *component) UpdateUser(ctx context.Context, realmName, userID string, us
 		}
 	}
 
-	var revokeAccreditations = keycloakb.IsUpdated(user.FirstName, oldUserKc.FirstName,
+	// Update in DB user for extra infos
+	// Store user in database
+	var userInfosUpdated = keycloakb.IsUpdated(user.BirthLocation, oldDbUser.BirthLocation,
```
Comment (fratt-elca) [inline pkg/management/component.go:475]: birthLocation is not a DEP mandatory field

File: pkg/management/component.go [hunk 2]
```diff
@@ -470,7 +470,16 @@ func (c *component) UpdateUser(ctx context.Context, realmName, userID string, us
...
+		user.Nationality, oldDbUser.Nationality,
```
Comment (fratt-elca) [inline pkg/management/component.go:476]: nationality is not a DEP mandatory field

File: pkg/management/component.go [hunk 3]
```diff
@@ -470,7 +470,16 @@ func (c *component) UpdateUser(ctx context.Context, realmName, userID string, us
...
+		user.IDDocumentType, oldDbUser.IDDocumentType,
```
Comment (fratt-elca) [inline pkg/management/component.go:477]: idDocumentType is not a DEP mandatory field

File: pkg/management/component.go [hunk 4]
```diff
@@ -470,7 +470,16 @@ func (c *component) UpdateUser(ctx context.Context, realmName, userID string, us
...
+		user.IDDocumentNumber, oldDbUser.IDDocumentNumber,
+		user.IDDocumentExpiration, oldDbUser.IDDocumentExpiration,
```
Comment (fratt-elca) [inline pkg/management/component.go:479]: idDocumentExpiration is not a DEP mandatory field

File: pkg/management/component.go [hunk 5]
```diff
@@ -470,7 +470,16 @@ func (c *component) UpdateUser(ctx context.Context, realmName, userID string, us
...
+		user.IDDocumentCountry, oldDbUser.IDDocumentCountry)
```
Comment (fratt-elca) [inline pkg/management/component.go:480]: idDocumentCountry is not a DEP mandatory field

Comment (fratt-elca) [inline pkg/management/component.go:480]: idDocumentCountry is not a DEP mandatory field

**Review by fratt-elca — CHANGES_REQUESTED** [review]
Only the DEP fields should void the accreditations :
- firstName
- lastName
- gender
- birthDate
- idDocumentNumber

PS1 : Sorry, the JIRA was not detailed enough, it's my fault.

PS2 : When we will implement the "Attribute management" and "Accreditations management" User Stories, this will be dynamic and each type of accreditations will require their own list of fields, and will be lost if one of those fields is changed. But for now, it's hardcoded for DEP because we're on production with that one.

**Review by fratt-elca — APPROVED** [review]
It works.

**Review by harture — APPROVED** [review]

**Pull request merged**